meson.build

```diff
@@ -129,10 +129,20 @@ conf.set('HAVE_STDBOOL_H', cc.has_header('stdbool.h'))
 conf.set('HAVE_MEMORY_H', cc.has_header('memory.h'))
 
 # Functions
-conf.set('HAVE_ALIGNED_ALLOC', cc.has_function('aligned_alloc', prefix: '#include <stdlib.h>'))
-conf.set('HAVE__ALIGNED_MALLOC', cc.has_function('_aligned_malloc', prefix: '#include <malloc.h>'))
-conf.set('HAVE_MEMALIGN', cc.has_function('memalign', prefix: '#include <stdlib.h>\n#include <malloc.h>'))
-conf.set('HAVE_POSIX_MEMALIGN', cc.has_function('posix_memalign', prefix: '#include <stdlib.h>'))
+if cc.has_function('memalign', prefix: '#include <stdlib.h>\n#include <malloc.h>')
+  conf.set('HAVE_MEMALIGN', 1)
+elif cc.has_function('_aligned_malloc', prefix: '#include <malloc.h>')
+  conf.set('HAVE__ALIGNED_MALLOC', 1)
+elif not (host_system == 'windows')
+  # These tests have proven unreliable on MSYS2 because meson falls
+  # back to testing compiler builtins instead of what we asked for.
```

Comment:
This comment is misleading since it implies that aligned malloc functions are not compiler built-ins whereas they almost always are. I would amend this to say:

```
Don't test these two on Windows because when building with GCC on MSYS2, Meson incorrectly detects these as being available even though they're not.
```

This, IMO is more descriptive and accurate.